### PR TITLE
Update to React 14 / ReactRouter 1.0

### LINKED ISSUE #1714: Upgrade to react v0.14
http://facebook.github.io/react/blog/2015/10/07/react-v0.14.html

### PR DESCRIPTION
**Pull request by aweiksnar**
:warning: ~~Not ready to merge yet; I'm still finishing up the todos in 1714, and doing lots of QA~~
:hourglass: Wanted to open this one up early since it touches almost every file and is a big diff

- This is a minimal upgrade path to React 14 / React Router 1.0
  - React 14 http://facebook.github.io/react/blog/2015/10/07/react-v0.14.html
  - React Router 1.0 https://github.com/rackt/react-router/blob/master/CHANGES.md#v100
- Upgrades deps to React 14 compatible versions 
- Closes #1714

### CODE REVIEW

File: app/collections/preview.cjsx [hunk 1]
```diff
@@ -27,20 +27,18 @@ module?.exports = React.createClass
...
       <div className="collection">
         <p className="title">
-          <Link to="collection-show" params={@collectionParams()}>
+          <Link to="/collections/#{owner}/#{name}">
             {@props.collection.display_name}
           </Link>
           {' '}by{' '}
           {if @state.owner
-            <Link className="user-profile-link" to="user-profile" params={name: @state.owner.login}>
+            <Link className="/users/#{@state.owner.login}" to="user-profile">
```
Comment (parrish) [inline app/collections/preview.cjsx]: Probably should be

```coffeescript
<Link className="user-profile-link" to="/users/#{@state.owner.login}">
```

Comment (aweiksnar) [inline app/collections/preview.cjsx]: Oops Yeah absolutely, just updated!

**Comment by parrish** [comment]
I fixed all (as far as I can tell anyway) of the remaining router changes in #1985.

A few issues I've seen so far.

- React complains a lot about a `div` inside a `p` somewhere.  Can't track that down though
- Admin pages don't render yet
- Intermittent `setState during render` errors, primarily in Talk

Just wanted to double check, but it looks like app/collections/home.cjsx isn't used.

**Comment by aweiksnar** [comment]
Yeah just removed collections/home.cjsx :+1: I'll look into those other warnings too. thanks for taking a look at this

**Comment by aweiksnar** [comment]
Admin page rendering and `p > div` nesting warnings should be all cleared up now

**Comment by parrish** [comment]
I've resolved every warning/error I can track down.  I've seen a `setState on an unmounted component` error a few times, but can't figure out how to reproduce it (it's probably some sort of race condition)

**Comment by parrish** [comment]
(Fixed one more)

**Comment by aweiksnar** [comment]
This should be good to go now, thanks for all the help & testing :bow: Added back in subdirectory deploys in 893afa1f7d99e59ccbdfd8438e3907938fd3aa8e - let me know if that looks cool to everyone or also any other issues that come up

**Comment by parrish** [comment]
:shipit: :sunglasses:

**Comment by brian-c** [comment]
I think all the merge conflicts are stuff from me, I'm fixing 'em now.

**Pull request merged**